src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -482,14 +482,21 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
                         data_type = r[1]
                         precision = 0
                         scale = 0
+                        internal_size = 0
                     else:
                         data_type = data_type_parts[0]
-                        precision = int(data_type_parts[1].split(",")[0])
-                        scale = int(data_type_parts[1].split(",")[1][:-1])
+                        if "," in data_type_parts[1]:
+                            precision = int(data_type_parts[1].split(",")[0])
+                            scale = int(data_type_parts[1].split(",")[1][:-1])
+                            internal_size = 0
+                        else:
+                            precision = 0
+                            scale = 0
+                            internal_size = int(data_type_parts[1][:-1])
```

Comment:
Curious: how is this related to the `StringType` change? 

Also, test coverage.

Reply:
The size of a string column comes in result metadata inside the `internal_size` field. So we use this to create `StringType()` on the client

Reply:
I couldn't find a way to specify string column max length in dataframe reader format so I'll remove this section and default column size to 0